Improve Contributing, Coding, and Pull Request Guidelines (#6157)
microsoft/MixedRealityToolkit-Unity: Documentation/Contributing/PullRequests.md

# Pull Requests

## Prerequisites
If you haven't contributed to a Microsoft project before, you may be asked to sign a [contribution license agreement](https://cla.microsoft.com/). 

> A comment in the PR will let you know if you do.

Reviewer: ```suggestion
An automatically generated comment in the PR will let you know if you do.
```

> [!IMPORTANT]
> If you are a Microsoft employee and are not a member of the [Microsoft organization on GitHub](https://github.com/Microsoft), please link your Microsoft and GitHub accounts on corpnet by visiting [Open Source at Microsoft](https://opensource.microsoft.com/) before you start your pull request. There's some process stuff you'll need to do ahead of time.

## Creating a Pull Request
When you are ready to submit a pull request, [create a pull request](https://github.com/microsoft/MixedRealityToolkit-Unity/compare/mrtk_development...mrtk_development?expand=1) targeting the [mrtk_development](https://github.com/microsoft/mixedrealitytoolkit-unity/tree/mrtk_development) branch.

> Read the guidelines and ensure your pull request meets the guidelines.

Reviewer: ```suggestion
Ensure the pull request meets the guidelines below.
```

* Make sure to reference any Issue / Feature Request or Task the PR relates to.
* Check the pull request contains only files / changes related to the PR.
* Check documentation is up to date and included. Check all public fields have comments.
* If adding a new feature, check that tests are included to validate the feature (see [UnitTests](UnitTests.md)).

> * If fixing a bug, write a test to verify the bug fix.

Reviewer: If any items have been renamed, ensure to search both MRTK code and documentation for impacted references

> The project maintainers will review your changes. We aim to review all changes within three business days. Please address any review comments, push to your topic branch, and post a comment letting us know that there's new stuff to review.

Reviewer: Should we address how tagging PRs or more importantly assigning right people to look at them?

> > [!NOTE]

Reviewer: honestly change to IMPORTANT

> All PR's submitted to the project will also be vetted according to the [MRTK coding standards guide](CodingGuidelines.md), so please review these before submitting your PR to ensure a smooth process.

## Pull Request Guidelines

> These guidelines are based off of the [Google's engineering practices](https://google.github.io/eng-practices/review/developer/small-cls.html).

Reviewer: ```suggestion
These guidelines are based off of [Google's engineering practices](https://google.github.io/eng-practices/review/developer/small-cls.html).
```

### Keep pull requests small
Smaller PRs are reviewed more quickly and thoroughly, are less likely to introduce bugs, easier to roll back, and easier to merge.

> Pull requests should be small enough that an engineer could review it in under 30 minutes. Try to make a minimal change that addresses just one thing. If you must create a large PR, split it into several PRs that go into either your local branch, or a feature branch of MRTK. Make sure any assets (e.g. fbx, obj files) are small.

Reviewer: I'd actually say, in general, avoid adding media assets to MRTK (e.g fbx, png files) and instead re-use existing content. If it is necessary to add unique assets, make sure any assets are small

### Tests should be added in the same PR as your fix / feature, except for emergencies

> Tests are the best way to ensure changes do not regress existing code, but it is also easy to forget about tests as people write new features. Requiring that they go in with your PR are a great way to ensure that tests get written.

Reviewer: nit: emphasis on tests for both features AND bug fixes
```suggestion
Tests are the best way to ensure changes do not regress existing code, but it is also easy to forget about tests when submitting pull requests. Requiring that they go in with your PR are a great way to ensure that tests get written.
```

> Every feature and bugfix should have tests associated with it. If you do not have the expertise or time to write a test, create an issue to write the tests, and mark them for “Consider for Current Iteration”.

Reviewer: ```suggestion
Every feature and bug fix should have tests associated with it. If you do not have the expertise or time to write a test, create an issue to write the tests, and mark them with label **Consider for Current Iteration**.
```

### Documentation should be added in the same pull request as a fix / feature

> Most developers look first at documentation, not code, when understanding how to use a feature. Ensuring documentation is up to date make it much easier for people to consume MRTK. Requiring documentation goes in with a pull request is a great way to ensure you write documentation for each feature.

Reviewer: ```suggestion
Most developers look first at documentation, not code, when understanding how to use a feature. Ensuring documentation is up to date makes it much easier for people to consume and rely MRTK.  Documentation should always be bundled with the related pull to ensure items remain up-to-date and consistent.
```
